spec/mongoid/association/referenced/belongs_to/proxy_spec.rb

```diff
@@ -1383,6 +1383,8 @@ class C
     end
 
     it "uses the correct explicit id" do
+      pending "https://jira.mongodb.org/browse/MONGOID-5089"
```

Comment:
MONGOID-5089 was fixed. Why are we reintroducing pending here? Was this perhaps an incorrectly resolved merge conflict?

Reply:
Changes in this PR un-fix [MONGOID-5089](https://jira.mongodb.org/browse/MONGOID-5089). We can reopen and re-triage it. Since it was for a very specific use case, it seems reasonable to sacrifice 5089 in favour of this one. What do you think?

Reply:
No, it doesn't make sense to reintroduce a bug to fix another bug. Please deliver a proper fix that covers both.

Reply:
We were not aware that [MONGOID-5089](https://jira.mongodb.org/browse/MONGOID-5089) was reported by a customer, the ticket was created by a team member. Thank you for pointing out that you were affected by it, we will reconsider the proposed solution.

Reply:
https://jira.mongodb.org/browse/MONGOID-5075 which is a dupe of https://jira.mongodb.org/browse/MONGOID-5089 was reported by a customer.